util/request_source.go

```diff
@@ -73,13 +81,27 @@ func IsRequestSourceInternal(reqSrc *RequestSource) bool {
 func (r *RequestSource) GetRequestSource() string {
 	// if r.RequestSourceType is not set, it's mostly possible that r.RequestSourceInternal is not set
 	// to avoid internal requests be marked as external(default value), return unknown source here.
-	if r == nil || r.RequestSourceType == "" {
+	if r == nil || (len(r.RequestSourceType) == 0 && len(r.ExplicitRequestSoureType) == 0) {
 		return SourceUnknown
 	}
+	appendType := func(list []string) []string {
+		if len(r.ExplicitRequestSoureType) > 0 {
+			list = append(list, r.ExplicitRequestSoureType)
+			return list
+		}
+		return list
+	}
 	if r.RequestSourceInternal {
-		return InternalRequest + r.RequestSourceType
+		labels := []string{InternalRequest, r.RequestSourceType}
+		labels = appendType(labels)
+		if len(r.ExplicitRequestSoureType) > 0 {
+			labels = append(labels, r.ExplicitRequestSoureType)
+		}
+		return strings.Join(labels, "_")
 	}
-	return ExternalRequest + r.RequestSourceType
+	labels := []string{ExternalRequest, r.RequestSourceType}
```

Comment:
if RequestSourceType is "" && ExplicitRequestSoureType is not "", 
 It will join a string like `external__XXX` which with double `_`, 
 does it make sense?